src/renderer/features/governance/aggregates/unlock.ts

```diff
@@ -60,6 +98,96 @@ sample({
   }),
 });
 
+sample({
+  clock: unlockConfirmAggregate.output.formSubmitted,
+  source: {
+    unlockData: $unlockData,
+    chain: networkSelectorModel.$governanceChain,
+    wrappedTxs: $wrappedTxs,
+  },
+  filter: ({ unlockData, chain, wrappedTxs }) => {
+    return Boolean(unlockData) && Boolean(chain) && Boolean(wrappedTxs);
+  },
+  fn: ({ unlockData, chain, wrappedTxs }) => ({
+    event: {
+      signingPayloads: wrappedTxs!.map((tx, index) => ({
+        chain: chain!,
+        account: unlockData!.shards[index],
+        signatory: unlockData!.signatory,
+        transaction: tx!,
+      })),
+    },
+    step: Step.SIGN,
+  }),
+  target: spread({
+    event: signModel.events.formInitiated,
+    step: stepChanged,
+  }),
+});
+
+sample({
+  clock: signModel.output.formSubmitted,
+  source: {
+    unlockData: $unlockData,
+    chain: networkSelectorModel.$governanceChain,
+    multisigTxs: $multisigTxs,
+    coreTxs: $coreTxs,
+    wrappedTxs: $wrappedTxs,
+    step: $step,
+  },
+  filter: (unlockData) => {
+    return (
+      !!isStep(unlockData.step, Step.SIGN) &&
+      !!unlockData.unlockData &&
+      !!unlockData.wrappedTxs &&
+      !!unlockData.coreTxs &&
+      !!unlockData.chain
+    );
+  },
+  fn: (unlockData, signParams) => {
+    return {
+      event: {
+        ...signParams,
+        chain: unlockData.chain!,
+        account: unlockData.unlockData!.shards[0],
+        signatory: unlockData.unlockData!.signatory,
+        description: unlockData.unlockData!.description,
+        coreTxs: unlockData.coreTxs!,
+        wrappedTxs: unlockData.wrappedTxs!,
+        multisigTxs: unlockData.multisigTxs || [],
+      },
+      step: Step.SUBMIT,
+    };
+  },
+  target: spread({
+    event: submitModel.events.formInitiated,
+    step: stepChanged,
+  }),
+});
+
+sample({
+  clock: delay(submitModel.output.formSubmitted, 2000),
+  source: $step,
+  filter: (step) => isStep(step, Step.SUBMIT),
+  target: flowFinished,
+});
+
+sample({
+  clock: submitModel.$submitStep,
+  source: {
+    chunks: unlockModel.$claimSchedule,
+    unlockData: $unlockData,
+  },
+  filter: ({ unlockData }, { step }) => !!unlockData && submitUtils.isSuccessStep(step),
+  fn: ({ chunks, unlockData }) => {
+    return chunks.filter(
+      (chunk) =>
+        chunk.type === UnlockChunkType.CLAIMABLE && unlockData!.shards.some((shard) => shard.address !== chunk.address),
```

Comment:
I think here better to add braces and return (here we have `=>` and return value on two different lines)